src/ParallelAlgorithms/Events/ObserveTime.hpp

```diff
@@ -0,0 +1,123 @@
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Reduction.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace dg {
+namespace Events {
+template <typename ObservationValueTag, typename EventRegistrars>
+class ObserveTime;
+
+namespace Registrars {
+template <typename ObservationValueTag>
+using ObserveTime =
+    ::Registration::Registrar<Events::ObserveTime, ObservationValueTag>;
+}  // namespace Registrars
+
+template <typename ObservationValueTag,
+          typename EventRegistrars =
+              tmpl::list<Registrars::ObserveTime<ObservationValueTag>>>
+class ObserveTime;  // IWYU pragma: keep
+
+/*!
+ * \ingroup DiscontinuousGalerkinGroup
+ * \brief %Prints the time of observation in the simulation.
+ *
+ * Prints reduction quantities:
+ * - `ObservationValueTag`
+ *
+ * \warning Currently, only one reduction observation event can be
+ * triggered at a given observation value.  Causing multiple events to run at
+ * once will produce unpredictable results.
+ */
+template <typename ObservationValueTag, typename EventRegistrars>
+class ObserveTime : public Event<EventRegistrars> {
+ private:
+  using ReductionData = Parallel::ReductionData<
+      Parallel::ReductionDatum<std::string, funcl::AssertEqual<>>>;
+
+ public:
+  /// \cond
+  explicit ObserveTime(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(ObserveTime);  // NOLINT
+  /// \endcond
+
+  using options = tmpl::list<>;
+  static constexpr OptionString help =
+      "Prints the time of observation in the simulation.\n"
+      "\n"
+      "Writes reduction quantities:\n"
+      " * ObservationValueTag\n"
+      "\n"
+      "Warning: Currently, only one reduction observation event can be\n"
+      "triggered at a given observation value.  Causing multiple events to\n"
+      "run at once will produce unpredictable results.";
+
+  ObserveTime() = default;
+
+  using observed_reduction_data_tags =
+      observers::make_reduction_data_tags<tmpl::list<ReductionData>>;
+
+  using argument_tags = tmpl::list<ObservationValueTag>;
+
+  template <typename Metavariables, typename ArrayIndex,
+            typename ParallelComponent>
+  void operator()(
+      const db::const_item_type<ObservationValueTag>& observation_value,
+      Parallel::ConstGlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/,
+      const ParallelComponent* const /*meta*/) const noexcept {
+    // Send data to reduction observer
+    auto& local_observer =
+        *Parallel::get_parallel_component<observers::Observer<Metavariables>>(
+             cache)
+             .ckLocalBranch();
+
+    std::string info_to_print =
```

Comment:
Would it be hard to add a timestamp either here or in the action that actually does the printing? Would be useful for long simulations to print [TIMESTAMP]: Time: #### instead of just the time, so you could also get a sense of how fast the run was running